### ISSUE TITLE
[requested feature] Consultation or Team Mode?

### ISSUE THREAD

**Issue by notruck** [gbtami/pychess-variants#42]
Could we have a "team mode" as a future option please?

If selected, this mode would let players choose and specify exactly who is on their team. I'm not sure whether a mid-game invite should be an option, it's probably best to have everyone join before the game starts. 

This new mode should be casual, and not affect anyone's ratings. 

Once in the game, they will each be able to add visualizations to the board, which only their team-mates can see. 

Ideally, any of the team members would be able to draw color arrows, mark highlights, make moves, and grant more time to the opponent team. This includes offering and accepting draws, and resigning on behalf of the whole team. If more than one member on the side-to-move were to attempt a move, whichever reaches the server first will register. 

At the end of the game, anyone can propose a rematch. The new game will either ask everyone if they would like to join, or just automatically include the same players as before, now with the colors reversed. 

In this mode, the chat window probably needs to be private as well for each team. I'm not sure how best to integrate private chats with public kibitz from spectators. 

I hope such features can potentially be useful for Bughouse play to some extent if/when that variant becomes available, although these days the preferred mode of communication for Bughouse coordination seems to be over voice chat.

Thanks!

**Comment by gbtami** [comment]
Good idea! I suggest one change though. As in real life consultancy games, only the camp leader should be able to take moves while game is ongoing.
![IMG_20191016_095516](https://user-images.githubusercontent.com/479964/66900003-43373780-effc-11e9-8991-23005ed4db0e.jpg)
